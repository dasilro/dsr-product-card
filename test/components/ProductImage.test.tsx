import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard, ProductImage } from '../../src/components';
import { product2 } from '../data/product';

describe('ProductImageu', () => {
  test('mostrar el product image', () => {
    const wrapper = renderer.create(
      <ProductImage img=".\test.png" className="custom-class" />
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('mostrar el product card con imagen', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>
        {() => <ProductImage img={product2.img}></ProductImage>}
      </ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
