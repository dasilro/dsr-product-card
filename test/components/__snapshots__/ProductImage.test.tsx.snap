// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImageu mostrar el product card con imagen 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffe-mug.png"
  />
</div>
`;

exports[`ProductImageu mostrar el product image 1`] = `
<img
  alt="Product"
  className="productImg custom-class"
  src=".\\\\test.png"
/>
`;
